components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3688,26 +3684,29 @@ where
                 .snapshot_recovery_maybe_finish_wait_apply(/* force= */ true);
         }
 
-        let mut meta = self.ctx.store_meta.lock().unwrap();
-        let is_region_initialized_in_meta = meta
-            .regions
-            .get(&region_id)
-            .map_or(false, |region| is_region_initialized(region));
-        if !is_initialized && is_region_initialized_in_meta {
-            let region_in_meta = meta.regions.get(&region_id).unwrap();
-            error!(
-                "peer is destroyed inconsistently";
-                "region_id" => region_id,
-                "peer_id" => self.fsm.peer_id(),
-                "peers" => ?self.region().get_peers(),
-                "merged_by_target" => merged_by_target,
-                "is_initialized" => is_initialized,
-                "is_region_initialized_in_meta" => is_region_initialized_in_meta,
-                "start_key_in_meta" => log_wrappers::Value::key(region_in_meta.get_start_key()),
-                "end_key_in_meta" => log_wrappers::Value::key(region_in_meta.get_end_key()),
-                "peers_in_meta" => ?region_in_meta.get_peers(),
+        (|| {
+            fail_point!(
+                "before_destroy_peer_on_peer_1003",
+                self.fsm.peer.peer_id() == 1003,
+                |_| {}
             );
-        }
+        })();
+        let mut meta = self.ctx.store_meta.lock().unwrap();
+        let is_latest_initialized = {
```

Comment:
if it's already initialized now, seems we can't call `meta.readers.remove(&region_id)` and others as well, otherwise the new split region's item is removed

Reply:
In the situation that `destroy_peer` is called by an uninitialized peer while it's already initialized in the region `meta`, I only find the case where the uninitialized peer is destroyed by a larger peer message during the splitting region happens in another thread by now. In this case, the newly inserted region or peer would be finally destroyed by the larger peer id message.

Skipping destroying the uninitialized peer seems to be an option too, maybe it's safer?